lib/node-repo.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+const githubClient = require('./github-client')
+
+const resolveLabels = require('./node-labels').resolveLabels
+
+function resolveLabelsThenUpdatePr (options) {
+  githubClient.pullRequests.getFiles({
+    user: options.owner,
+    repo: options.repo,
+    number: options.prId
+  }, (err, res) => {
+    if (err) {
+      return console.error(`! ${prInfoStr(options)} Error retrieving files`, err)
+    }
+
+    const filepathsChanged = res.map((fileMeta) => fileMeta.filename)
+    updatePrWithLabels(options, resolveLabels(filepathsChanged))
+  })
+}
+
+function updatePrWithLabels (options, labels) {
+  if (!labels) {
+    return
+  }
+
+  githubClient.issues.edit({
+    user: options.owner,
+    repo: options.repo,
+    number: options.prId,
+    labels: labels
```

Comment:
Little worried about this, in a theoretical race would this overwrite the existing labels? e.g. if a collaborator opens a PR that adds labels from the start?